pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/SyntheticStageGraphListener.java

```diff
@@ -98,23 +94,19 @@ private void attachTag(FlowNode currentNode, String syntheticContext) {
         }
     }
 
-    @Extension
-    public static class FlowExecutionListenerImpl extends FlowExecutionListener {
-        @Override
-        public void onRunning(FlowExecution execution) {
-            attachGraphListener(execution);
-        }
-
-        @Override
-        public void onResumed(FlowExecution execution) {
-            attachGraphListener(execution);
-        }
-
-        private void attachGraphListener(FlowExecution execution) {
-            if (execution != null && !execution.isComplete()) {
-                execution.addListener(new SyntheticStageGraphListener());
+    private boolean isDeclarativeRun(@Nonnull FlowExecution execution) {
+        try {
+            FlowExecutionOwner owner = execution.getOwner();
+            if (owner != null && owner.getExecutable() instanceof WorkflowRun) {
+                if (((WorkflowRun) owner.getExecutable()).getAction(ExecutionModelAction.class) != null) {
```

Comment:
Could just use `Run` or even `Actionable`—no need for `WorkflowRun`, unless you are already relying on that elsewhere here.